packages/internal/src/validateSchema.ts

```diff
@@ -31,9 +34,45 @@ export function validateSchemaForDirectives(
           // skip validation for redwood query and currentUser
           if (!(isRedwoodQuery || isCurrentUserQuery)) {
             const hasDirective = field.directives?.length
+
             if (!hasDirective) {
               validationOutput.push(`${fieldName} ${fieldTypeName}`)
             }
+
+            // we want to check that the requireAuth directive roles argument value
+            // is a string or an array of strings
```

Comment:
The problem with this check DT, is that it makes this PR a breaking change. 

So in earlier templates we used to generate directives like this:

```
    deleteContact(id: Int!): Contact! @requireAuth(roles: ["ADMIN"])
```

What would happen in this situation? 

I totally understand why you added the extra validation here, but I think it would be better to not have this just for backwards compatibility with the directives that we've already shipped in the last 10 versions or so.

Reply:
(but I may be confused about the change, can you confirm that it isn't breaking?)

Reply:
I'll test

```ts
    deleteContact(id: Int!): Contact! @requireAuth(roles: ["ADMIN"])
```

but the rule is that the roles can be a string or an array of string so that should be fine.

```ts
    deleteContact(id: Int!): Contact! @requireAuth(roles: ["ADMIN"])
```

```ts
    deleteContact(id: Int!): Contact! @requireAuth(roles: "ADMIN")
```

Would be the same, I think.

Reply:
Added a test for the uppercase Mutation 

```ts
export const schema = gql`
  type Query {
    deleteTodo(id: Int!): Todo! @requireAuth(roles: ["ADMIN"])
  }
`
```